Gems/ImGui/Code/Source/ImGuiManager.cpp

```diff
@@ -401,60 +401,65 @@ bool ImGuiManager::OnInputChannelEventFiltered(const InputChannel& inputChannel)
     const InputDeviceId& inputDeviceId = inputChannel.GetInputDevice().GetInputDeviceId();
 
-    // Handle Keyboard Hotkeys
-    if (InputDeviceKeyboard::IsKeyboardDevice(inputDeviceId) && inputChannel.IsStateBegan())
-    {
-        // Cycle through ImGui Menu Bar States on Home button press
-        if (inputChannelId == InputDeviceKeyboard::Key::NavigationHome)
-        {
-            ToggleThroughImGuiVisibleState();
-        }
+    bool consumeEvent = false;
 
-        // Cycle through Standalone Editor Window States
-        if (inputChannel.GetInputChannelId() == InputDeviceKeyboard::Key::NavigationEnd)
+    // Handle Keyboard Inputs
+    if (InputDeviceKeyboard::IsKeyboardDevice(inputDeviceId))
+    {
+        // Handle Keyboard Hotkeys
+        if (inputChannel.IsStateBegan())
         {
-            if (gEnv->IsEditor() && m_editorWindowState == DisplayState::Hidden)
+            // Cycle through ImGui Menu Bar States on Home button press
+            if (inputChannelId == InputDeviceKeyboard::Key::NavigationHome)
             {
-                ImGuiUpdateListenerBus::Broadcast(&IImGuiUpdateListener::OnOpenEditorWindow);
+                ToggleThroughImGuiVisibleState();
             }
-            else
+
+            // Cycle through Standalone Editor Window States
+            if (inputChannel.GetInputChannelId() == InputDeviceKeyboard::Key::NavigationEnd)
             {
-                m_editorWindowState = m_editorWindowState == DisplayState::Visible
-                                          ? DisplayState::VisibleNoMouse
-                                          : DisplayState::Visible;
+                if (gEnv->IsEditor() && m_editorWindowState == DisplayState::Hidden)
+                {
+                    ImGuiUpdateListenerBus::Broadcast(&IImGuiUpdateListener::OnOpenEditorWindow);
+                }
+                else
+                {
+                    m_editorWindowState = m_editorWindowState == DisplayState::Visible
+                                              ? DisplayState::VisibleNoMouse
+                                              : DisplayState::Visible;
+                }
             }
         }
-    }
-
-    // Handle Keyboard Modifier Keys
-    if (InputDeviceKeyboard::IsKeyboardDevice(inputDeviceId))
-    {
-        if (inputChannelId == InputDeviceKeyboard::Key::ModifierShiftL
-            || inputChannelId == InputDeviceKeyboard::Key::ModifierShiftR)
-        {
-            io.KeyShift = inputChannel.IsActive();
-        }
-        else if (inputChannelId == InputDeviceKeyboard::Key::ModifierAltL
-                 || inputChannelId == InputDeviceKeyboard::Key::ModifierAltR)
-        {
-            io.KeyAlt = inputChannel.IsActive();
-        }
-        else if (inputChannelId == InputDeviceKeyboard::Key::ModifierCtrlL
-                 || inputChannelId == InputDeviceKeyboard::Key::ModifierCtrlR)
+        // Handle Keyboard Modifier Keys
+        else
```

Comment:
I might just not be reading the scopes here correctly, but it looks like the behavior has changed here in that we're no longer handling modifier keys or setting io.KeysDown if (inputChannel.IsStateBegan()) - removing the else would retain the same behavior as before?

Reply:
You would be correct the behaviour changed by mistake, will remove the else here.